pkg/nsx/services/subnet/subnet.go

```diff
@@ -308,8 +308,9 @@ func (service *SubnetService) UpdateSubnetSetStatus(obj *v1alpha1.SubnetSet) err
 }
 
 func (service *SubnetService) ListSubnetID() sets.String {
-	subnetSet := service.SubnetStore.ListIndexFuncValues(common.TagScopeSubnetCRUID)
-	return subnetSet
+	subnets := service.SubnetStore.ListIndexFuncValues(common.TagScopeSubnetCRUID)
+	subnetSets := service.SubnetStore.ListIndexFuncValues(common.TagScopeSubnetSetCRUID)
+	return subnets.Union(subnetSets)
 }
```

Comment:
Can we remove `ListSubnetID()` and change `Cleanup()` to something like:
```
func (service *SubnetService) Cleanup() error {
	subnets := service.SubnetStore.List()
	log.Info("cleaning up subnet", "count", len(subnets))
	for _, subnet := range subnets {
		nsxSubnet, _ := subnet.(model.VpcSubnet)
		err := service.DeleteSubnet(nsxSubnet)
		if err != nil {
			return err
		}
	}
	return nil
}
```
because I think it's strange to firstly get the ids from store and secondly get objects from the store by ids...